tests/foreman/virtwho/utils.py

```diff
@@ -115,31 +148,34 @@ def get_rhsm_logfile():
 
     The following keys will be included:
     :[error]: how many error message exist, the expected value is 0.
-    :[reporter_id]: it can be defined by virtwho configure.
-    :[interval_time]: it can be defined by virtwho configure.
-    :[hypervisor_id]: it can be defined by virtwho configure.
-    :[hypervisor_name]: get to know the hypervisor's hostname.
-    :[guest_id]: get to know the guest_id.
+    :[hypervisor_name]: the hypervisor display name in Content Hosts.
+    :[guest_name]: the guest display name in Content Hosts.
     """
-    ret, stdout = runcmd('cat /var/log/rhsm/rhsm.log')
     data = dict()
+    mapping = list()
+    entry = None
+    guest_name, guest_uuid = get_guest_info()
+    ret, stdout = runcmd('cat /var/log/rhsm/rhsm.log')
     data['error'] = len(re.findall(r'\[.*ERROR.*\]', stdout))
-    if "virtwho.main DEBUG" in stdout:
-        res = re.findall(r"reporter_id='(.*?)'", stdout)
-        if len(res) > 0:
-            data['reporter_id'] = res[0].strip()
-        res = re.findall(r"infinite loop with(.*?)seconds", stdout)
-        if len(res) > 0:
-            data['interval_time'] = int(res[0].strip())
-        res = re.findall(r'[^\s]*"hypervisorId": "(.*?)"', stdout)
-        if len(res) > 0:
-            data['hypervisor_id'] = res[0].strip()
-        res = re.findall(r'[^\s]*"name": "(.*?)"', stdout)
-        if len(res) > 0:
-            data['hypervisor_name'] = res[0].strip()
-        res = re.findall(r'[^\s]*guestId": "(.*?)"', stdout)
-        if len(res) > 0:
-            data['guest_id'] = res[0].strip()
+    data['guest_name'] = guest_name
+    for line in stdout.split('\n'):
+        if line:
+            if line[0].isdigit():
+                if entry:
+                    mapping.append(_parse_entry(entry))
+                entry = '{'
+                continue
+            if entry:
+                entry += line
+            else:
+                mapping.append(_parse_entry(entry))
```

Comment:
In my rough attempt, that `else` was at level of `for`. The idea was to catch JSON that happens to be in last line of log file.

Reply:
okay, make sense to me.